x-pack/plugins/endpoint/public/applications/endpoint/store/policy_details/selectors.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createSelector } from 'reselect';
+import { PolicyDetailsState } from '../../types';
+
+export const selectPolicyDetails = (state: PolicyDetailsState) => state.policyItem;
+
+export const selectPolicyIdFromParams: (state: PolicyDetailsState) => string = createSelector(
```

Comment:
You might not need this. In the component, you can use one of `react-router-dom` `useParams()` hook ( [ref](https://reacttraining.com/react-router/web/api/Hooks/useparams) ) to get back the URL params.

Reply:
or `useRouteMatch()` hook (also from `react-router-dom` 😬

Reply:
@paul-tavares I'm currently using the selector inside the middleware to get the id when `userNavigatedToPage` is fired.

https://github.com/elastic/kibana/pull/59951/files#diff-b7089396eb7c6a8054fbeafc623b9261R16

Wouldn't I need to be in the component itself to use the hook?

Reply:
Yeah, it would need to be done in a component. It just feels wrong to be parsing the route outside of react-router. 
React-router has [`matchPath()`](https://reacttraining.com/react-router/web/api/matchPath) which allows you to pass a URL (aka: matched path) in along with a Router's Path and get back the route match which has any route param that was used in that route. If we go this way, then I would also suggest centralizing the router `path`'s as constants somewhere so we have a single definition for them (which btw - could also help in building up URL for use in `href`'s by using the react-router `generatePath()` function.

Reply:
@paul-tavares created this to address the above: elastic/endpoint-app-team#268